Add CHANGELOG warning about EC generation with keysize (#115)
corretto/amazon-corretto-crypto-provider: CHANGELOG.md

# Changelog

## 1.5.0 (Unreleased)
### Breaking Change Warning
In accordance with our [versioning policy](https://github.com/corretto/amazon-corretto-crypto-provider/blob/master/VERSIONING.rst),

> we post warnings of upcoming changes which may cause compatibility issues.

Reviewer: ```suggestion
we post warnings of upcoming changes that may cause compatibility issues.
```

Reviewer 2: ```suggestion
we post warnings of upcoming changes that might cause compatibility issues.
```

As always, we expect that these changes will not impact the vast majority of consumers and can be picked up automatically provided you have good unit and integration changes.

> Starting in ACCP vesion 1.6.0, EC key pair generation will throw an `InvalidParameterException` if initialized to a keysize not on the following list.

Reviewer 2: ```suggestion
Starting in ACCP vesion 1.6.0, EC key pair generation will throw an `InvalidParameterException` if initialized to a keysize that is not in the following list.
```

For these explicit sizes (only), the behavior will remain unchanged with ACCP selecting the corresponding "secp*r1" curve

> (which, in these cases, is also the corresponding NIST Prime Curve).

Reviewer: It might be more clear to show the mapping and state that it has not changed.

> For these explicit sizes (only), the behavior will remain unchanged with ACCP selecting the corresponding "secp*r1" curve
> (which, in these cases, is also the corresponding NIST Prime Curve).

Reviewer 2: ```suggestion
For these explicit sizes (only), ACCP behavior is unchanged. ACCP selects the the "secp*r1" curve that corresponds to the value. (For these values, its also the corresponding NIST prime curve).
```
-or- (since we're explaining behavior that _isn't_ changing:
For these explicit sizes (only), ACCP behavior is unchanged. ACCP selects the the "secp*r1" curve that corresponds to the value.

**Supported keysize values:**
* 192
* 224
* 256
* 384
* 521

> This means that the following code will start failing.

Reviewer: ```suggestion
This means that the following code will start failing because it requests a keysize that is not on the list.
```

> This means that the following code will start failing.

Reviewer 2: ```suggestion
This means that the following code will start failing.
```
Therefore, the following code fails because it specifies an unsupported value.

Do we really need all of this explanation? These are the supported values. Everything else fails. Do we need a code sample for that?

```java
KeyPairGenerator kg = KeyPairGenerator.getInstance("EC");
kg.initialize(160); // Throws an InvalidParameterException
```

> We are making this change because the "SunEC" provider does not document its curve selection process for sizes other than those listed above and does not promise that the curve selected will remain consistent.

Reviewer: ```suggestion
We are making this change because the "SunEC" provider does not document its curve selection process for sizes other than those listed above and does not promise that it will continue to use the same curve selection process.
```

> With no consistency guarantees, it is not possible for developers to use
> [KeyPairGenerator.initialize(int keysize)](https://docs.oracle.com/javase/8/docs/api/java/security/KeyPairGenerator.html#initialize-int-)

Reviewer 2: ```suggestion
Without aconsistency guarantee, developers can't use
[KeyPairGenerator.initialize(int keysize)](https://docs.oracle.com/javase/8/docs/api/java/security/KeyPairGenerator.html#initialize-int-)
```

safely (regardless of whether ACCP is used or not).

**We strongly recommend using**
**[KeyPairGenerator.initialize(AlgorithmParameterSpec params)](https://docs.oracle.com/javase/8/docs/api/java/security/KeyPairGenerator.html#initialize-java.security.spec.AlgorithmParameterSpec-)**
**with**
**[ECGenParameterSpec](https://docs.oracle.com/javase/8/docs/api/java/security/spec/ECGenParameterSpec.html)**
**to generate EC keys.**

> From versions 1.2.0 through 1.5.0, ACCP *always* selects the corresponding "secp*r1" curve.

Reviewer: ```suggestion
From versions 1.2.0 through 1.5.0, ACCP selects the corresponding "secp*r1" curve for any keysize requested.
```
